### DISCUSSION TITLE
Discharge vs Charging Slots

### DISCUSSION THREAD

**Discussion by dieswarte** [springfall2008/batpred#876] [Q&A]
Hi, I'm on IOG, with 2 x 9.5 KWh batteries and a Gen 2 HY 5.0 inverter. The 6-hour cheap slot from 23:30 - 05:30 is sufficient to always fill up the 2 batteries with cheap electricity and then I would run them throughout the day.

I have noticed over the last few weeks, that Batpred will always start to discharge the battery at some point during the night, to empty the batteries before the cheap slot. However, the slots that it would do it in is always from whatever time in the evening until 23:00. Then from 23:00 - 23:30 there is nothing in the plan which I presume it runs the battery in eco mode and just supplies to the house on demand, before starting to charge the batteries up from 23:30 onwards.

However, what I have noticed for a while now is that it looks like I'm inevitably then using electricity from the grid between 23:00 - and 23:30 at the peak rate.

Is there a specific setting I can do so it discharges to 23:30 rather than 23:00? I'm not sure why this is the case but was wondering if you need a slot of the inverter doing nothing between discharging and charging?

**Comment by iainfogg** [comment]
Hi @dieswarte - if the battery level has been taken too low and it's running out before 23:30 I'd adjust Best SOC Keep to a higher level so there's more buffer for the 23:00-23:30 period.

I've got the same setup as you and I've got it set to 1.7 if that helps?

@springfall2008 did explain to me a while ago why it doesn't discharge down to 23:30 and instead stops at 23:00 but I've forgotten the answer, sorry!

**Comment by gcoan** [comment]
Another option you could do is to put a relative cost rate adjustment for export_override in apps.yaml to make it ‘appear’ more profitable to discharge from 23:30.

But start with best soc keep

**Comment by gcoan** [comment]
There have been recent fixes by Trefor to address this issue.  Reopen if still a problem
